compiler/rustc_macros/src/query.rs

```diff
@@ -286,12 +254,10 @@ fn add_query_description_impl(query: &Query, impls: &mut proc_macro2::TokenStrea
                 #expr
             }
 
-            #try_load_from_disk
+            const TRY_LOAD_FROM_DISK: Option<fn(QueryCtxt<'tcx>, SerializedDepNodeIndex) -> Option<Self::Value>>
+                = Some(crate::plumbing::try_load_from_disk::<Self::Value>);
```

Comment:
I'm not really fond of referencing code from `rustc_query_impl` here.
Could we make `TRY_LOAD_FROM_DISK` a `bool`, and have `rustc_query_impl` pass the function pointer, or does it cause an issue with the where clause on `V: Decodable`?
Other possibility: add a `cache` modifier to be read by a macro in `rustc_query_impl` and avoid instantiating an uncallable function?

Reply:
r=me if impossible / too impractical.

Reply:
> Could we make `TRY_LOAD_FROM_DISK` a `bool`, and have `rustc_query_impl` pass the function pointer, or does it cause an issue with the where clause on `V: Decodable`?

The where clause is exactly why I ended up with the current function, yeah. I didn't try to hard to fix it - I'll take a look this evening and see what I can do.

Reply:
Other possibility: get rid of the `TRY_LOAD_FROM_DISK` constant, and use the same device as we use with `no_hash/hash_results`: a word in the modifiers, and a macro to select on it.

Reply:
> Other possibility: get rid of the TRY_LOAD_FROM_DISK constant, and use the same device as we use with no_hash/hash_results: a word in the modifiers, and a macro to select on it.

this is an excellent idea, done :)